### ISSUE TITLE
wireguard-go: occasional tai64n test flake

### ISSUE THREAD

**Issue by crawshaw** [tailscale/tailscale#174]
```
--- FAIL: TestMonotonic (0.85s)
tai64n_test.go:21: Whitening insufficient
FAIL
FAIL	github.com/tailscale/wireguard-go/tai64n	0.875s
```

Also, move this issue to WireGuard/wireguard-go later.

**Comment by dshynkev** [comment]
I will try to build a definitive test case, but I think this might be due to the test process getting scheduled away. That `tai64n_test.go` relies on that lines (1) and (2) execute within ~16.7ms of each other, which, I imagine, may sometimes fail to happen:
```go
for i := 0; i < 500; i++ {
	next := Now()  // (2)
	if next.After(old) {
		t.Error("Whitening insufficient")
	}
	time.Sleep(time.Duration(whitenerMask)/time.Nanosecond + 1)
	next = Now()
	if !next.After(old) {
			// ...
	}
	old = next  // (1)
}
```

**Comment by crawshaw** [comment]
The correct thing to do here may be an API change. Instead of `func Now() Timestamp` the package could expose `func Stamp(t time.Time) Timestamp` which would be deterministic.

As a first step towards that, you could add a Stamp function with an implementation of Now passing in the value of time.Now() and rewrite the tests to use Stamp.

**Comment by dshynkev** [comment]
Sounds good. I postponed more experimentation with the scheduler to tomorrow, but that's the only source of problems I can see here anyway, so I'll gladly just fix the API instead.

**Comment by dshynkev** [comment]
The fix has been merged upstream (Wireguard/wireguard-go@f60b391). This issue should go away when `tailscale/wireguard-go` is rebased.
